src/scrapers/unionbank.js

```diff
@@ -0,0 +1,283 @@
+        return trs.map(async (tr) => ({
+            id: tr.getAttribute('id'),
+            innerTds: await pageEvalAll(tr, 'td', [], (tds) => {
+                return tds
+            })
+        }));
+    });
+
+    for (const txnRow of transactionsRows) {
+        handleTransactionRow(txns, transactionsTableHeaders, txnRow, txnType);
+    }
+    return txns;
+}
+
+async function isNoTransactionInDateRangeError(page) {
+  const hasErrorInfoElement = await elementPresentOnPage(page, '.errInfo');
+  if (hasErrorInfoElement) {
+    const errorText = await page.$eval('.errInfo', (errorElement) => {
+      return errorElement.innerText;
+    });
+    return errorText === NO_TRANSACTION_IN_DATE_RANGE_TEXT;
+  }
+  return false;
+}
+
+async function getTransactionsTableHeaders(page, tableTypeId) {
+    return await pageEvalAll(page, `#WorkSpaceBox #${tableTypeId} tr[class='header'] th`, [], (ths) => {
+        return ths.map(async (th, index) => (
+            ths[th.innerText] = index
+        ));
+    });
+}
+
+async function chooseAccount(page, accountId) {
+    const hasErrorInfoElement = await elementPresentOnPage(page, 'select#ddlAccounts_m_ddl');
+    if (hasErrorInfoElement.offsetParent !== null) {
+        await dropdownSelect(page, 'select#ddlAccounts_m_ddl', accountId);
+    }
+}
+
+async function searchByDates(page, startDate) {
+    await dropdownSelect(page, 'select#ddlTransactionPeriod', '004');
+    await waitUntilElementFound(page, 'select#ddlTransactionPeriod');
+    await fillInput(
+        page,
+        'input#dtFromDate_textBox',
+        startDate.format(DATE_FORMAT),
+    );
+    await clickButton(page, 'input#btnDisplayDates');
+    await waitForNavigation(page);
+}
+
+async function getAccountNumber(page) {
+    const selectedSnifAccount = await page.$eval('#ddlAccounts_m_ddl option[selected="selected"]', (option) => {
+        return option.innerText;
+    });
+
+    return selectedSnifAccount.replace('/', '_');
+}
+
+async function expandTransactionsTable(page) {
+    const hasExpandAllButton = await elementPresentOnPage(page, "a[id*='lnkCtlExpandAll']");
+    if (hasExpandAllButton) {
+        await clickButton(page, "a[id*='lnkCtlExpandAll']");
+    }
+}
+
+async function scrapeTransactionsFromTable(page) {
+    const pendingTxns = await extractTransactionsFromTable(page, PENDING_TRANSACTIONS_TABLE_ID, TRANSACTION_STATUS.PENDING);
+    const completedTxns = await extractTransactionsFromTable(page, COMPLETED_TRANSACTIONS_TABLE_ID, TRANSACTION_STATUS.COMPLETED);
+    const txns = [
+        ...pendingTxns,
+        ...completedTxns,
+    ];
+    return convertTransactions(txns);
+}
+
+async function getAccountTransactions(page) {
+    try {
+        // await Promise.race([
+        //     waitUntilElementFound(page, '#ctlActivityTable', false),
+        //    // waitUntilElementFound(page, '.errInfo', false),
+        // ]);
+
+        if (await isNoTransactionInDateRangeError(page)) {
```

Comment:
@dratler I usually prefer to see consts initialized for await statements before using their values - it's more readable and makes it easier to debug
```suggestion
        const noTransactionInRangeError = await isNoTransactionInDateRangeError(page);
        if (noTransactionInRangeError) {
```